csrc/optimization/alias_analysis.h

```diff
@@ -0,0 +1,22 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <unordered_map>
+
+#include <fusion.h>
+#include <ir/interface_nodes.h>
+
+namespace nvfuser::optimization {
+
+// Maps aliases (e.g. fusion outputs) to their sources (e.g. fusion inputs).
+using AliasAnalysisResult =
+    std::unordered_map<const TensorView*, const TensorView*>;
```

Comment:
I know this is a WIP.

should we use a disjoint-set as the value, rather than a single map from TV -> TV.

i.e. In the case below, we should be able to recognize that all tvs in the fusion points to the same buffer?
```
  const std::vector<int64_t> in_shape({2, 3, 4});
  const std::vector<int64_t> inter_shape({2, 12});
  const std::vector<int64_t> out_shape({24});

  TensorView* in = makeContigConcreteTensor(in_shape);
  fusion.addInput(in);
  TensorView* inter = reshape(in, in_shape, inter_shape);
  TensorView* out = reshape(inter, inter_shape, out_shape);
  fusion.addOutput(out);
```

I guess this could make bookkeeping tricky for cases where we are slicing through the tensor and create an alias that doesn't cover the whole buffer.